horizon/middleware_manager.py

```diff
@@ -0,0 +1,144 @@
+import asyncio
+from copy import deepcopy
+import json
+from datetime import datetime
+
+from app.database import CREDENTIALS
+from horizon.middleware_api import get_players_online, authenticate_async, authenticate, get_active_games, get_recent_stats, get_recent_game_history
+from horizon.parsing.uya_game import uya_map_parser, uya_time_parser, uya_gamemode_parser
+
+from app.schemas.schemas import (
+    Pagination,
+    UyaPlayerOnlineSchema,
+    UyaGameOnlineSchema,
+    DeadlockedPlayerOnlineSchema,
+    DeadlockedGameOnlineSchema,
+)
+
+
+class MiddlewareManager:
+    def __init__(self, game:str, players_online_poll_interval:int=60, token_poll_interval:int=3600):
+        """
+        Class to manager all middleware calls and polling.
+
+        :param players_online_poll_interval: Interval time in seconds to poll the players online API
+        :param token_poll_interval: Interval time in seconds to refresh the token
+        """
+        self._players_online_poll_interval = players_online_poll_interval
+        self._token_poll_interval = token_poll_interval
+        self._players_online = []
+        self._games_online = []
+
+        self._protocol: str = CREDENTIALS[game]["horizon_middleware_protocol"]
+        self._host: str = CREDENTIALS[game]["horizon_middleware_host"]
+        self._horizon_app_id: int = CREDENTIALS[game]["horizon_app_id"]
+        self._horizon_username: str = CREDENTIALS[game]["horizon_middleware_username"]
+        self._horizon_password: str = CREDENTIALS[game]["horizon_middleware_password"]
+
+        self._token: str = authenticate(
+            protocol=self._protocol,
+            host=self._host,
+            username=self._horizon_username,
+            password=self._horizon_password
+        )
+
+    async def refresh_token(self) -> None:
+        # Periodically refresh token so we don't get 401
+        while True:
+            self._token = await authenticate_async(
+                protocol=self._protocol,
+                host=self._host,
+                username=self._horizon_username,
+                password=self._horizon_password
+            )
+            await asyncio.sleep(self._token_poll_interval)
+
+    async def poll_players_online(self) -> None:
```

Comment:
I would rename `poll_players_online` since this handles more than updating the players list.